structuretoolkit/visualize.py

```diff
@@ -145,17 +145,16 @@ def plot3d(
 
 
 def _get_frame(cell):
-    vertices = np.array([[0, 0, 0], [0, 0, 1], [0, 1, 1], [0, 1, 0], [0, 0, 0]])
-    return [
-        f(vertices + v) @ cell
-        for f in [lambda x: x, lambda x: np.roll(x, 1, axis=-1)]
-        for v in [0, [1, 0, 0]]
-    ]
+    lines_dz = np.stack(np.meshgrid(*3 * [[0, 1]], indexing="ij"), axis=-1)
+    all_lines = np.reshape(
+        [np.roll(lines_dz, i, axis=-1) for i in range(3)], (-1, 2, 3)
+    )
```

Comment:
```suggestion
    lines_dz = np.stack(np.meshgrid(*3 * [[0, 1]], indexing="ij"), axis=-1)
    # eight corners of a unit cube, paired as four z-axis lines
    
    all_lines = np.reshape(
        [np.roll(lines_dz, i, axis=-1) for i in range(3)], (-1, 2, 3)
    )
    # All 12 two-point lines on the unit square
```

TBH I don't find either this PR or the existing code particularly intelligible. The new version is only two lines, which I like, but I'd rather give a hint what all the numpy magic is doing.

Reply:
Me neither. Just as you pointed out, except for the fact that it's very short I don't particularly like it either.